core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -216,7 +216,8 @@ class LogManager(logDirs: Array[File],
       maxProducerIdExpirationMs = maxPidExpirationMs,
       scheduler = scheduler,
       time = time,
-      brokerTopicStats = brokerTopicStats)
+      brokerTopicStats = brokerTopicStats,
+      logDirFailureChannel = logDirFailureChannel)
```

Comment:
I see. In general, we should do our best to avoid making production code more brittle due to tests. I would personally remove all the default arguments from `Log.apply`. By far most of the usages are from `LogTest`, so we can simply add a method within that class to make the migration easier.